### ISSUE TITLE
[Desktop] Youtube lag

### ISSUE THREAD

**Issue by ze0ss** [brave/brave-browser#10507]
The video player takes a long time to go fullscreen on www.youtube.com, after the fullscreen buttonin bottom right is pressed or F key is pressed.
It takes about 12 or more seconds.

## Steps to Reproduce
   1. Open youtube.com, and open a full HD video
   2. Check video quality is 1080p, preferably 4K or higher
   3. Click the fullscreen icon at bottom right, or press the F key once

## Actual result:
The player takes 12+ seconds to switch to full screen.

## Expected result:
It must take the least amount of time, 2 seconds at the most

## Reproduces how often: 
Easily reproduced

## Brave version (brave://version info)

Brave | 1.10.97 Chromium: 83.0.4103.116 (Official Build) (64-bit)
-- | --
Revision | 8f0c18b4dca9b6699eb629be0f51810c24fb6428-refs/branch-heads/4103@{#716}
OS | Windows 10 OS Version 1909 (Build 18363.900)
JavaScript | V8 8.3.110.13
Flash | (Disabled)
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36
Command Line | "C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe" -incognito --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --sync-url=https://no-thanks.invalid --enable-features=PasswordImport,AutoupgradeMixedContent,PassiveMixedContentWarning,MixedContentSiteSetting,DnsOverHttps,WebUIDarkMode --disable-features=AutofillServerCommunication,VideoPlaybackQuality,TextFragmentAnchor,SmsReceiver,TabHoverCards,LookalikeUrlNavigationSuggestionsUI,AutofillEnableAccountWalletStorage,AllowPopupsDuringPageUnload,NotificationTriggers --flag-switches-begin --load-media-router-component-extension=1 --flag-switches-end --enable-audio-service-sandbox --disable-sync
Executable Path | C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe
Profile Path | C:\Users\user2\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default
Variations | 5e3a236d-4113a79e

## Version/Channel Information:
This issue happens on the version I have installed, which is the stable branch, I didn't try to reproduce this issue on any other build like nightly or beta because I seldom Install unstable builds :)

## Other Additional Information:
This issue is not resolved after enabling/disabling any extensions or brave rewards.

## Miscellaneous Information:
Device HW info:
CPU - i5 9th Gen notebook series 
GPU - 1) Intel iGPU 2) Nvidia dGPU
RAM - 8 GB
Screen resolution - 1920*1080

All drivers are updated to latest versions, and windows 10 is updated too.

Please note: This issue is completely absent in other browsers, especially non-chromium based ones, like Firefox, it takes less than a second to go full screen or come back in stock firefox.

https://imgur.com/rn9Lkji

**Comment by srirambv** [comment]
What is the behaviour on Chrome? Is this on primary display or secondary display?

**Comment by abh-oss01** [comment]
On chrome it's a little bit faster than brave, 
and I only have one display, it's a laptop.
This issue is observed only on google chrome and brave, basically chromium based browsers.

**Comment by srirambv** [comment]
If its happening on Chrome as well may be an upstream issue that Chromium team will fix it and we will need to wait for it in a Chromium bump.

**Comment by abh-oss01** [comment]
No, not even an issue at this point, this is happening since a long time on chrome, like I said, I have no idea how issue fixing from upstream works, but I just wish to point out to you that this "issue" has been there for a very long time, and there seems to be no headway from google too, so I don't actually know what happens next, on brave.. etc
Like, would the team try to fix this issue themselves or something like that?
Because google doesn't seem to be interested in this at all, it seems.

**Comment by srirambv** [comment]
cc: @rebron @bsclifton  to decide on priority so it can be fixed or wait for upstream

**Comment by abh-oss01** [comment]
Thanks a lot, looking forward for the same.

**Comment by segovia-no** [comment]
I also have the same problem, after changing the video quality and scrubbing the video for a while, then the playback suffers a lot, it drops about 40 to 60% of the frames at any resolution

**Comment by bsclifton** [comment]
This issue hasn't had an update in a while - it's met the criteria for being stale. The issue is going to be closed for now. If this is still a problem, please leave a comment or re-open the issue. Thanks!
